src/llvm-julia-licm.cpp

```diff
@@ -275,6 +274,17 @@ struct JuliaLICM : public JuliaPassContext {
                     if (valid) {
                         ++HoistedAllocation;
                         moveInstructionBefore(*call, *preheader->getTerminator(), MSSAU, SE);
+                        IRBuilder<> builder(preheader->getTerminator());
+                        auto obj_i8 = builder.CreateBitCast(call, Type::getInt8PtrTy(call->getContext(), call->getType()->getPointerAddressSpace()));
+                        // Note that this alignment is assuming the GC allocates 16-byte aligned memory
+                        auto clear_obj = builder.CreateMemSet(obj_i8, ConstantInt::get(Type::getInt8Ty(call->getContext()), 0), call->getArgOperand(1), Align(16));
```

Comment:
Is there any TBAA or `invariant.group` data that we have that we should try to add?

Reply:
I can't think of any useful TBAA metadata (we're scribbling over every field simultaneously without regard for types) to add. I don't know about invariant.group, but the gc_alloc_obj call itself is annotated with noalias, so that might be sufficient for alias analysis?